pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -200,3 +297,143 @@ func (k *klusterletStatusController) sync(ctx context.Context, controllerContext
 	)
 	return nil
 }
+
+func buildKubeClientWithSecret(secret *corev1.Secret) (kubernetes.Interface, error) {
+	defer func() {
+		for key := range secret.Data {
+			os.Remove(path.Join("/tmp", key))
+		}
+	}()
+
+	for key, data := range secret.Data {
+		if err := ioutil.WriteFile(path.Join("/tmp", key), data, 0644); err != nil {
+			return nil, err
+		}
+	}
+	restConfig, err := clientcmd.BuildConfigFromFlags("", path.Join("/tmp", "kubeconfig"))
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(restConfig)
+}
+
+func createSelfSubjectAccessReviews(
+	ctx context.Context,
+	kubeClient kubernetes.Interface,
+	selfSubjectAccessReviews []authorizationv1.SelfSubjectAccessReview) (bool, *authorizationv1.SelfSubjectAccessReview, error) {
+	for _, subjectAccessReview := range selfSubjectAccessReviews {
+		ssar, err := kubeClient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, &subjectAccessReview, metav1.CreateOptions{})
+		if err != nil {
+			return false, &subjectAccessReview, err
+		}
+		if !ssar.Status.Allowed {
+			return false, &subjectAccessReview, nil
+		}
+	}
+	return true, nil, nil
+}
+
+func getBootstrapSelfSubjectAccessReviews() []authorizationv1.SelfSubjectAccessReview {
```

Comment:
It would be awesome to be able to generate these somehow, but let's not bother for now.